Formula/g/ghc.rb

```diff
@@ -38,73 +29,54 @@ class Ghc < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "python@3.11" => :build
-  depends_on "sphinx-doc" => :build
-  depends_on macos: :catalina
+  depends_on "xz" => :build
+  depends_on "libffi"
```

Comment:
I'm not seeing where this is used

```
   /opt/homebrew/Cellar/ghc/9.6.2/lib/ghc-9.6.2/lib/aarch64-osx-ghc-9.6.2/libffi.dylib (ghc)
Dependencies with no linkage:
  libffi
```

Reply:
I've removed it, it was added in #137249, but I don't know why ... let's try without it